src/lib/pages/common/infiniteScroll.js

```diff
@@ -0,0 +1,24 @@
+// TODO: find correct place for this function
+/** @type {import('svelte/action').Action}  */
+export default function scroll(node) {
+  const options = { threshold: 1 }; // threshold: 1 = entire element must be visible to trigger
+
+  const callback = (entries) => {
+    entries.forEach((e) => {
+      if (e.isIntersecting) {
+        e.target.dispatchEvent(new CustomEvent('infiniteScroll'));
+      }
+    });
+  };
+
+  const observer = new IntersectionObserver(callback, options);
+
+  observer.observe(node);
+
+  return {
+    destroy() {
+      // destroying the observer when the component is no longer loaded
+      observer.destroy();
```

Comment:
I see an exception in the console when reaching the end of the results:

```
Uncaught (in promise) TypeError: t.destroy is not a function
```

I guess it’s [`disconnect`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/disconnect):

```suggestion
      observer.disconnect();
```

Reply:
Good catch! I'll change it to the correct function.